### PR TITLE
Add scalafix rule for renaming `assert` to `expect`

### PR DESCRIPTION
**Pull request by zainab-ali**
This is a follow up to #69 . The `RenameAssertToExpect` rule renames calls to weaver `assert` to `expect`.

As part of the release, it would be great to add upgrade notes to the documentation and a [scalafix migration](https://github.com/scala-steward-org/scala-steward/blob/main/docs/scalafix-migrations.md#adding-migration-rules-to-scala-steward).

### CODE REVIEW

**Comment by Baccata** [comment]
I'm not fully versed in scalafix migrations, but considering none of this code is wired in the sbt-build, does it mean that the `scalafix` folder is meant to be a self-contained project and scala-steward is able to build it on the fly ?

File: scalafix/build.sbt [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+      TargetAxis.resolve(input, Compile / scalaVersion).value
+  )
+  .defaultAxes(
+    rulesCrossVersions.map(VirtualAxis.scalaABIVersion) :+ VirtualAxis.jvm: _*
+  )
+  .jvmPlatform(
+    scalaVersions = Seq(V.scala212),
+    axisValues = Seq(TargetAxis(scala3Version)),
+    settings = Seq()
+  )
```
Comment (kubukoz) [inline scalafix/build.sbt:84]: Can you explain the need for the custom `TargetAxis` please? (also I suppose it has something to do with why this is mixing 2.12 and 3.x)

**Review by kubukoz — COMMENTED** [review]
Are we missing the definition of `RenameAssertToExpect`?

**Comment by zainab-ali** [comment]
The `scalafix` directory is completely separate from the weaver build. It's intended to be temporary: we can delete this directory as soon as the `assert` / `expect` change is released.

A lot of the files (`build.sbt` and `TargetAxis`) were created by the `scalafix` Giter8 template. They let us run local tests against different Scala versions with `sbt tests/test`.

Since we don't need to maintain them past the release, I haven't looked into them in detail.

## Scala steward migrations

The rule will be run by Scala steward, provided we add some config to the Scala steward repo.

The only necessary file for scala steward (or our users) to run the rule is the `RenameAssertToExpect.scala` file.

You can try this out locally on any project with the `sbt-scalafix` plugin:

```
sbt scalafixAll github:zainab-ali/typelevel-weaver-test/RenameAssertToExpect?assert-scalafix-rule
```

## The release process

When we instruct scala steward (or our users) to apply the rule, we'll reference a specific tag.

For example, if this rule is meant to be run as part of a `v1.0.0` release, we can tell users to run:

```
sbt scalafixAll github:typelevel/weaver-test/RenameAssertToExpect?v1.0.0
```

Since the file is referenced through a tag, we can delete the `scalafix` directory after the `v1.0.0` release.

On a related note, I'm looking into how we can use scala steward's [artifact migrations](https://github.com/scala-steward-org/scala-steward/blob/main/docs/artifact-migrations.md) to seamlessly migrate users from disney to typelevel. Scala steward would need to migrate the artifact and apply the `RenameAssertToExpect` rule in one go.

I don't think there are any other API changes we can mitigate through scalafix rules. We have changed the signature of `expect.same`, but this should be source compatible for most users.

**Review by Baccata — APPROVED** [review]

**Pull request merged**